backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/acct/api/AccountDimension.java

```diff
@@ -256,15 +269,18 @@ public Builder setAlias(final String alias)
 			return this;
 		}
 
-		public final Builder setSegmentValue(final AcctSegmentType segmentType, final Object value)
+		private Builder setSegmentValue(final AcctSegmentType segmentType, final Object value)
 		{
-			final Object valueConverted;
-
-			if(value instanceof Integer)
+			if (value instanceof Integer)
 			{
 				final int intValue = NumberUtils.asInt(value, 0);
 				segmentValues.put(segmentType, intValue);
 			}
+			else if (value instanceof Instant)
+			{
+				Optional.ofNullable(TimeUtil.asInstant(value))
+						.ifPresent(instantValue -> segmentValues.put(segmentType, instantValue));
```

Comment:
no need to use optional here, if value is null => `value instanceof Instant` will be false

Reply:
what we need instead, is to make sure we can handle the "null" string when reading dates from segmentValues